### DISCUSSION TITLE
HASSIO Backup and restore

### DISCUSSION THREAD

**Discussion by DavidRBailey** [zwave-js/zwave-js-ui#2557] [Q&A] [answered]
Short of setting up SSH on the host OS and accessing /mnt/data/supervisor/, is there a way to backup and restore the zwavejs2mqtt configuration and other related data?

**Comment by robertsLando** [comment]
> is there a way to backup and restore the zwavejs2mqtt configuration and other related data?

Sure! You can handle automatic scheduled backups from settings :)

**Reply by DavidRBailey** [reply]
Thanks.

Where is the ZWaveJS2MQTT data stored in the config directory, or is it pulling from somewhere else?

**Reply by robertsLando** [reply]
> Where is the ZWaveJS2MQTT data stored in the config directory, or is it pulling from somewhere else?

It stores everyhing on `store` directory

**Reply by DavidRBailey** [reply]
So, my original question stands– "Short of setting up SSH on the host OS and accessing /mnt/data/supervisor/, is there a way to backup and restore the zwavejs2mqtt configuration?"

Note the "restore" part. I know you have automated scheduled backups in the settings.

I'd also like to be able to get the backup files OFF the local storage in case of a total storage loss, like I've already experienced.

**Comment by DavidRBailey** [comment, marked as answer]
I've been poking around trying to understand how the HASSIO backup and restore works, and how it relates to the Z-Wave JS to MQTT backups.

Here's what I found, use at your own risk some or all of this may be wrong as I'm a relative noob to HA–

1. You have to configure and turn on the Z-Wave JS to MQTT backups, let it run, and then make a full HA system backup.
2. You'll want to store at least some of the HA system backups on an external storage device in case your HASSIO storage dies. (My new purpose-purchased high-end storage card did after a few months.)
3. The Z-Wave JS to MQTT backup files are stored in the HA system backup file.
4. The documentation just says to restore everything in the HA system backup, but if you don't want to, you're going to need to de-archive the HA system backup TAR file and the specific a0d7b954_zwavejs2mqtt.tar.gz file.
5. If you use a password to protect the backup, you're going to need [this python script](https://community.home-assistant.io/t/backup-snapshots-no-longer-decryptable/104048/12) to extract the files from the protected HA system backup file, as HA uses a proprietary encrypted TAR file format. (seriously?!?)
6. The Z-Wave JS to MQTT files are in the a0d7b954_zwavejs2mqtt folder.

Cheers.

**Reply by kpine** [reply]
HAOS backup functionality is documented in the HA docs. https://www.home-assistant.io/common-tasks/os/#backups. If you're using that method, all you have to do restore an add-on backup is upload the backup file.

If you are using HAOS add-on backups, is it necessary to also enable store backups? The add-on backup would include everything in zwavejs2mqtt's store. So if you just turn on NVM backups in z2m, and let HAOS take care of the rest, is it equivalent? Or does the z2m store backups provide something additional?

**Reply by robertsLando** [reply]
If the HAOS backup already backups data there is no reason to also enable store backup, just NVM backups should be enabled
